### DISCUSSION TITLE
opening page mouse cues demo

### DISCUSSION THREAD

**Discussion by alior101** [infer-actively/pymdp#61] [General]
Hi,
Where can I find the python scripts used to generate this cool mouse doing some epistemic magic gif demo ? 
Thanks
Lior

**Comment by conorheins** [comment]
Hi Lior,
Thanks for your interest -- as promised, you can find the notebook for the epistemic chaining demo here: https://github.com/infer-actively/pymdp/blob/master/docs/notebooks/cue_chaining_demo.ipynb

Note that I haven't included the actual code for making the gif; instead, I replaced the GIF code with static plots that show the trajectory of the rat agent over time, superimposed on the grid world. I did this because the code for making the .gif is quite messy and not really integral to the concepts explored in this demo.

I can nevertheless share the messy gif code with you / the PNG images I used for the different graphics, if you're interested in this demo for the sake of the animation code (which was done using `FuncAnimation` from matplotlib).

Best,
Conor

**Comment by alior101** [comment]
Thanks! I'll be happy to take a look at your messy code...
Again, thanks for the hard work!
